src/components/MyKiva/BadgesSection.vue

```diff
@@ -0,0 +1,127 @@
+<template>
+	<div class="tw-w-full tw-inline-flex tw-flex-wrap tw-justify-center tw-gap-2.5">
+		<div
+			v-for="badge in badgesArray"
+			:key="badge.fields.key"
+			class="badge-container tw-flex tw-flex-col tw-justify-between tw-p-1.5 tw-rounded"
+			:class="{
+				'tw-bg-white': badge.hasStarted,
+				'tw-border-4 tw-border-tertiary tw-border-dashed': !badge.hasStarted
+			}"
+		>
+			<span class="tw-text-base !tw-font-medium tw-text-center tw-mb-1">
+				{{ getBadgeTitle(badge) }}
+			</span>
+			<div
+				class="tw-p-1"
+				:class="{
+					'tw-grayscale': !badge.hasStarted
+				}"
+				style="height: 148px;"
+			>
+				<img
+					:src="getBadgeImgUrl(badge)"
+					class="tw-h-full tw-mx-auto"
+				>
+			</div>
+			<div class="tw-flex tw-flex-col tw-gap-0.5 tw-mt-2 tw-font-medium">
+				<span
+					v-if="badge.hasStarted"
+					class="tw-mx-auto"
+				>
+					Level {{ badge.level }}/5
+				</span>
+				<button
+					class="tw-text-action hover:tw-underline"
+					v-kv-track-event="[
+						'portfolio',
+						'click',
+						badge.hasStarted ? 'Continue' : 'Start this journey',
+						getBadgeTitle(badge),
+						badge.level
+					]"
+					@click="() => $emit('badge-clicked', badge)"
+				>
+					{{ badge.hasStarted ? 'Continue' : 'Start this journey' }}
+				</button>
+			</div>
+		</div>
+	</div>
+</template>
+
+<script setup>
+import { computed, toRefs } from 'vue';
+
+const WOMEN_KEY = 'womens-equality';
+const US_EQUALITY_KEY = 'us-economic-equality';
+const CLIMATE_KEY = 'climate-action';
+const REFUGEE_KEY = 'refugee-equality';
+const BASIC_NEEDS_KEY = 'basic-needs';
+
+const defaultBadges = [
+	WOMEN_KEY,
+	US_EQUALITY_KEY,
+	BASIC_NEEDS_KEY,
+	CLIMATE_KEY,
+	REFUGEE_KEY,
+];
+
+defineEmits(['badge-clicked']);
+
+const props = defineProps({
+	badgesData: {
+		type: Array,
+		default: () => ([])
+	},
+	userAchievements: {
+		type: Array,
+		default: () => ([])
+	}
+});
+
+const { badgesData, userAchievements } = toRefs(props);
+
+const badgesArray = computed(() => {
+	const badges = [];
+	if (badgesData.value.length > 0) {
+		defaultBadges.forEach(badgeKey => {
+			let badgeFound = badgesData.value.find(entry => entry.fields.key === `${badgeKey}-level-1`);
+			const userAchievement = userAchievements.value.find(entry => entry.id === badgeKey);
+
+			if (!userAchievement) {
+				badgeFound = {
+					...badgeFound,
+					hasStarted: false,
+					level: 0,
+				};
+			} else {
+				const hasStarted = userAchievement.status !== 'NO_PROGRESS';
+				const level = userAchievement.totalProgressToAchievement;
```

Comment:
Is this the correct level to show? I thought `totalProgressToAchievement` was the number of loans to that level of the badge. Here I wonder if instead we need the highest badge level from the achievement service that is completed.

Reply:
Nice call! I guess yeah, we need the level from the service for this... or to clarify where the level is coming from.

Reply:
We can use this as dummy and change it when having more data... I'll add a comment for that.

Reply:
Sounds good, add a `TODO` comment and add a ticket to the epic, thanks!